qui/clipboard.py

```diff
@@ -75,40 +77,110 @@ def my_init(self, loop=None, gtk_app=None):
         self.gtk_app = gtk_app
         self.loop = loop if loop else asyncio.get_event_loop()
 
-    def _copy(self, vmname: str = None):
+    def _copy(self, metadata: dict) -> None:
         ''' Sends Copy notification via Gio.Notification
         '''
-        if vmname is None:
-            with appviewer_lock():
-                with open(FROM, 'r', encoding='ascii') as vm_from_file:
-                    vmname = vm_from_file.readline().strip('\n')
-
-        size = clipboard_formatted_size()
-
-        body = _("Clipboard contents fetched from qube: <b>'{vmname}'</b>\n"
+        size = clipboard_formatted_size(metadata["sent_size"])
+
+        if metadata["malformed_request"]:
+            body = _("Malformed clipboard request received from qube: "
+                     "<b>{vmname}</b>!").format(vmname=metadata["vmname"])
+        elif metadata["qrexec_clipboard"] and \
+                metadata["sent_size"] >= metadata["buffer_size"]:
```

Comment:
Why is this not treated as oversized request? As in, why is this a separate elif?

Reply:
Because Microsoft Windows uses another path to send clipboard to global clipboard (qrexec instead of X11 gui daemon). And the patch to gui-daemon and gui-agent was getting extremely complex and big. Developing a mechanism for Microsoft Windows to notify oversized clipboards to GUIVM needs additional effort. So the GUIVM has no mechanism to detect of to detect Windows oversized requests.

I guess it would be OK to work on this again after official support for Microsoft Windows is restored.